### PR TITLE
Hyper-modeling sample

### PR DESCRIPTION
**Pull request by pmconne**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "@bentley/context-registry-client": "2.12.1",
     "@bentley/frontend-authorization-client": "2.12.1",
     "@bentley/geometry-core": "2.12.1",
+    "@bentley/hypermodeling-frontend": "2.11.1",
```
Comment (MichaelBelousov) [inline package.json:10]: why not 2.12.1? (master recently updated to 2.12.1)

Comment (pmconne) [reply package.json:10]: Added that dep a couple weeks ago on this branch. Will update.

Comment (pmconne) [reply package.json]: Updated.

**Comment by BaoTon** [comment]
It's a little unclear where the markers are - any way we can make this more obvious?

image

I'm sometimes confused on which side of the arrow I can click on.

Also not sure if this is a new/old issue  but switching between any sample and the viewer-only-2d-sample causes an error on this branch:
image

@tomdicarlo  - any ideas what may be causing this? It doesn't seem to be present in the current production build.

**Comment by pmconne** [comment]
@BaoTon The arrows are ugly graphics from MicroStation that I'd prefer we turned off in all the samples. I can turn them off in this one. Clicking on them does nothing - only clicking on the markers does anything.

The hypermodeling package draws the markers as canvas decorations. It can't display them behind the model. It could make them smaller based on distance from camera but I'm not sure how much that would help.

I'll look into the exception.

**Comment by Josh-Schifter** [comment]
When I start the sample, I'm not seeing any markers, no section, and the UI is all disabled.  Am I missing something?

image

**Comment by pmconne** [comment]
@Josh-Schifter pretty clearly - check the console? Did you `npm install`? There's a new dependency.

**Comment by Josh-Schifter** [comment]
>> The arrows are ugly graphics from MicroStation that I'd prefer we turned off in all the samples. I can turn them off in this one. 

Let's change the defaults so that we don't see the ugly graphics for the house in all the samples.  We can do that by special casing the house model in the ViewSetup class.

**Comment by Josh-Schifter** [comment]
@pmconne, I did install.  Nothing interesting in the console.

**Comment by BaoTon** [comment]
> @pmconne, I did install. Nothing interesting in the console.

I think you mean it's unclear how to interact with the sample? The markers are close to the design callouts. You'll know you can trigger the section when you see a 'pointer' on hover. I think this just has to do with how unclear the markers are.

**Comment by pmconne** [comment]
@BaoTon your screenshots are not displaying the markers either. The markers are round SVG images.

**Comment by Josh-Schifter** [comment]
@BaoTon, I think you might be right.  Blindly clicking around I did get an elevation hypermodel to work.
@pmconne, I think the markers are just invisible for both Bao and I.

@pmconne, when the sample starts, the view should look like the thumbnail (which looks great BTW).

**Comment by BaoTon** [comment]
> @BaoTon, I think you might be right. Blindly clicking around I did get an elevation hypermodel to work.
> @pmconne, I think the markers are just invisible for both Bao and I.
> 
> @pmconne, when the sample starts, the view should look like the thumbnail (which looks great BTW).

You'll need to run 'npm run build' before 'npm run start' to get your icons to show up. The markers are a lot more obvious.

image

**Comment by Josh-Schifter** [comment]
Much better now that I can see the markers!

When I click on the 'Section-Wall' marker, the view zooms in much too close.

**Comment by pmconne** [comment]
> When I click on the 'Section-Wall' marker, the view zooms in much too close.

Yes, that's a problem with the source data.
If you want, I can turn that marker off in the sample.

**Comment by pmconne** [comment]
@BaoTon @Josh-Schifter
I changed ViewSetup to hide the callouts.
The initial view now has the same section activated as in the thumbnail.
I hid the "Section-Wall" marker.
I have been unable to reproduce the exception Bao mentioned.

**Comment by BaoTon** [comment]
Looks like I just have local issues - seems to be happening on other branches for me as well. Can skip the error concern.

**Comment by pmconne** [comment]
@BaoTon @williamkbentley @Josh-Schifter are you awaiting something further from me?

File: src/frontend-samples/hypermodeling-sample/readme.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+* How to initialize and configure the @bentley/hypermodeling-frontend package.
+* How to toggle hyper-modeling in a viewport.
+
+## Description
+
+Computer-aided design and drafting (CADD) workflows often produce the following:
+
+* A 3d model of an infrastructure asset, such as a building;
+* A set of 2d section drawings produced by slicing the 3d model with a clippin plane; and
```
Comment (williamkbentley) [inline src/frontend-samples/hypermodeling-sample/readme.md:19]: `clippin plane`

**Review by BaoTon — APPROVED** [review]
LGTM

File: src/frontend-samples/hypermodeling-sample/HyperModelingUI.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import "common/samples-common.scss";
+import { ControlPane } from "common/ControlPane/ControlPane";
+import { SandboxViewport } from "common/SandboxViewport/SandboxViewport";
+import * as React from "react";
+import { assert, Id64String } from "@bentley/bentleyjs-core";
+import { IModelApp, ScreenViewport, ViewState } from "@bentley/imodeljs-frontend";
+import { SectionMarker } from "@bentley/hypermodeling-frontend";
```
Comment (williamkbentley) [inline src/frontend-samples/hypermodeling-sample/HyperModelingUI.tsx:11]: In problems pane: Cannot find module '@bentley/hypermodeling-frontend' or its corresponding type declarations.

See [Stefan's comments](https://github.com/imodeljs/frontend-sample-showcase/pull/160#issuecomment-792984872)

**Review by williamkbentley — CHANGES_REQUESTED** [review]

File: src/frontend-samples/hypermodeling-sample/HyperModelingUI.tsx (hunk 3 above)
Comment (pmconne) [reply src/frontend-samples/hypermodeling-sample/HyperModelingUI.tsx:11]: Fixed (this should be documented somewhere?)

File: src/frontend-samples/hypermodeling-sample/readme.md (hunk 2 above)
Comment (pmconne) [reply src/frontend-samples/hypermodeling-sample/readme.md]: Fixed.

**Review by williamkbentley — APPROVED** [review]
This is awesome

File: src/frontend-samples/hypermodeling-sample/HyperModelingUI.tsx (hunk 3 above)
Comment (williamkbentley) [reply src/frontend-samples/hypermodeling-sample/HyperModelingUI.tsx:11]: I believe there is a bit of info in Contributing.md, but could be clarified.

**Pull request merged**